tests/conftest.py

```diff
@@ -1,138 +1,369 @@
+    """Create and return a computer configured for Firecrest.
 
-def pytest_terminal_summary(
-    terminalreporter: TerminalReporter, exitstatus: int, config: pytest.Config
-):
-    """Called after all tests have run."""
-    data = config.stash.get("firecrest_requests", None)
-    if data is None:
-        return
-    terminalreporter.write(
-        yaml.dump(
-            {"Firecrest requests telemetry": data},
-            default_flow_style=False,
-            sort_keys=True,
-        )
+    Note, the computer is not stored in the database.
+    """
+
+    # create a temp directory and set it as the workdir
+    _scratch = tmpdir / "scratch"
+    _temp_directory = tmpdir / "temp"
+    _scratch.mkdir()
+    _temp_directory.mkdir()
+
+    Path(tmpdir / ".firecrest").mkdir()
+    _secret_path = Path(tmpdir / ".firecrest/secret69")
+    _secret_path.write_text("SECRET_STRING")
+
+    computer = orm.Computer(
+        label="test_computer",
+        description="test computer",
+        hostname="-",
+        workdir=str(_scratch),
+        transport_type="firecrest",
+        scheduler_type="firecrest",
+    )
+    computer.set_minimum_job_poll_interval(5)
+    computer.set_default_mpiprocs_per_machine(1)
+    computer.configure(
+        url=" https://URI",
+        token_uri="https://TOKEN_URI",
+        client_id="CLIENT_ID",
+        client_secret=str(_secret_path),
+        client_machine="MACHINE_NAME",
+        small_file_size_mb=1.0,
+        temp_directory=str(_temp_directory),
     )
+    return computer
+
+
+class MockFirecrest:
+    def __init__(self, firecrest_url, *args, **kwargs):
+        self._firecrest_url = firecrest_url
+        self.args = args
+        self.kwargs = kwargs
+
+        self.whoami = whomai
+        self.list_files = list_files
+        self.stat = stat_
+        self.mkdir = mkdir
+        self.simple_delete = simple_delete
+        self.parameters = parameters
+        self.symlink = symlink
+        self.checksum = checksum
+        self.simple_download = simple_download
+        self.simple_upload = simple_upload
+        self.compress = compress
+        self.extract = extract
+        self.copy = copy
+        self.submit = submit
+        self.poll_active = poll_active
+
+
+class MockClientCredentialsAuth:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
 
 
 @pytest.fixture(scope="function")
-def firecrest_server(
+def myfirecrest(
     pytestconfig: pytest.Config,
-    request: pytest.FixtureRequest,
     monkeypatch,
-    tmp_path: Path,
 ):
-    """A fixture which provides a mock Firecrest server to test against."""
-    config_path: str | None = request.config.getoption("--firecrest-config")
-    no_clean: bool = request.config.getoption("--firecrest-no-clean")
-    record_requests: bool = request.config.getoption("--firecrest-requests")
-    telemetry: RequestTelemetry | None = None
-
-    if config_path is not None:
```

Comment:
Why did you actually remove the option to add a custom config? Is this extra maintenance? It seems like a nice feature.

Reply:
This was used to run tests against an actual FirecREST server. The way it was done by deploying a docker image provided in [firecrest repo](https://github.com/eth-cscs/firecrest/tree/master/deploy/demo). However the docker image is not functional/maintained enough, I tried many times the docker image has some problem with storage endpoint. 
Therefore I thought to drop this feature for now, will retrieve later if the docker image is maintained again.

Reply:
For the record, now the server tests are also functional. 
afew doesn't pass only because there are afew bugs from FirecREST side that has to be resolved, this one is particularly blocking:
https://github.com/eth-cscs/firecrest/issues/205